crates/ra_parser/src/grammar/expressions/atom.rs

```diff
@@ -170,8 +170,28 @@ fn array_expr(p: &mut Parser) -> CompletedMarker {
     if p.eat(T![']']) {
         return m.complete(p, ARRAY_EXPR);
     }
+
+    // test first_array_member_attributes
+    // pub const A: &[i64] = &[
+    //    #[cfg(test)]
+    //    1,
+    //    2,
+    // ];
+    let first_member_has_attrs = p.at(T![#]);
+    attributes::outer_attributes(p);
+
     expr(p);
     if p.eat(T![;]) {
+        if first_member_has_attrs {
+            // test_err array_length_attributes
+            // pub const A: &[i64] = &[
+            //   #[cfg(test)]
+            //   1;
+            //   2,
+            // ];
+            p.error("removing an expression is not supported in this position");
```

Comment:
Should this be a *parse* error though? The error message sounds like it's from macro-expansion and cfg processing. 

Looks like we should just allow attributes on all expressions?

Reply:
Ah yes, you're right. It looks like the `rustc` parser allows arbitrary attributes here. I'll remove the error.